analysis/src/CompletionBackEnd.ml

```diff
@@ -1090,7 +1090,10 @@ let rec completeTypedValue (t : SharedTypes.completionType) ~env ~full ~prefix
       |> List.filter (fun (field : field) ->
              List.mem field.fname.txt seenFields = false)
       |> List.map (fun (field : field) ->
-             Completion.create field.fname.txt
+             let fname =
+               PrintType.printIdentLike ~allowUident:false field.fname.txt
+             in
+             Completion.create fname
```

Comment:
One idea is to not use the `name` field for this, but instead the `insertText` field. That'll make the name appear normal in the completion list, but it's the escaped identifier that'll be insterted. What do you think?

Reply:
We can also consider whether this should be done inside of `Completion.create` directly. Probably not though, as there are quite a few instances where it shouldn't be escaped even when there are exotic characters in there (like with variant payloads, `Some(whatver)`).